### PR TITLE
Grails 7 compatibility

### PR DESCRIPTION
**Pull request by matrei**
This PR updates the project to Java 17, Groovy 4 and Jakarta.
Hopefully the SNAPSHOT can be used for getting some more tests in `grails-gsp` update to pass.

### CODE REVIEW

**Comment by matrei** [comment]
Related: grails/grails-core#13555

File: build.gradle [hunk 1]
```diff
@@ -42,7 +52,7 @@ subprojects { project ->
     }
 
     repositories {
-        mavenLocal()
+        //mavenLocal()
```
Comment (sdelamo) [inline build.gradle]:
```suggestion
```

File: settings.gradle [hunk 2]
```diff
@@ -31,6 +31,6 @@ include 'grails-testing-support',
         'grails-web-testing-support',
         'grails-gorm-testing-support'
 
-include 'examples-demo33'
-project(":examples-demo33").projectDir = new File(settingsDir, "examples/demo33")
+//include 'examples-demo33'
```
Comment (sdelamo) [inline settings.gradle:34]: why comment this?

Comment (matrei) [reply settings.gradle:34]: @sdelamo The example test project does not currently work and blocks
progress. This commit disables it from the build temporarily
until conditions have changed.

**Review by sdelamo — APPROVED** [review]

**Pull request merged**